cliquet/tests/resource/test_viewset.py

```diff
@@ -277,9 +310,15 @@ def test_record_views_are_registered_in_cornice(self, service_class):
 
     @mock.patch('cliquet.resource.Service')
     def test_collection_methods_are_skipped_if_not_enabled(self, service_cls):
-        register_resource(self.resource, viewset=self.viewset, settings={
+        venusian_callback = register_resource(
+            self.resource, viewset=self.viewset)
+
+        context = mock.MagicMock()
+        context.registry.settings = {
             'cliquet.record_fake_put_enabled': False
-        })
+        }
+        context.config.with_package.return_value = context
```

Comment:
Are you sure that it works without mocks ? I mean that with_package actually returns you that?